src/main/java/org/folio/rs/controller/PubSubEventController.java

```diff
@@ -40,17 +40,19 @@ public class PubSubEventController implements PubSubHandlersApi {
   @Override
   public ResponseEntity<String> pubSubHandlersLogRecordEventPost(@Valid PubSubEvent pubSubEvent) {
     if (Objects.nonNull(pubSubEvent)) {
+      log.debug("pubSubHandlersLogRecordEventPost pubSubEvent:{}",pubSubEvent);
       RequestEvent requestEvent = null;
       try {
         var logEventType = pubSubEvent.getLogEventType();
         if (isItemCheckedId(logEventType)) {
           returnItemService.returnItem(pubSubEvent.getItemBarcode());
         } else {
           var payload = MAPPER.writeValueAsString(pubSubEvent.getPayload());
-          if (isPagedRequestCreated(logEventType, payload)) {
+          log.info("pubSubHandlersLogRecordEventPost payload:{}",payload);
+          if (Objects.nonNull(pubSubEvent.getPayload()) && isPagedRequestCreated(logEventType, payload)) {
```

Comment:
I am not sure this is correct solution just to check the payload for null. It allows us to avoid error in the log, but it does not solve the source of the problem when PubSubEvent contains null instead of payload, though all required data to create payload are present in the message from mod-pubsub module.

Reply:
Yes that's correct , but I observed for checkout-event the payload send by mod-pubsub is always null which is send to this particular endpoint , and this method does not process **checkout event** ( it only process paged event ) so while checking if it is a paged event if we add this null check from mod-remote-storage side we can avoid this .

Reply:
So in addition to that , even if mod-pubusb sending null payload for Checkout event , as we are not processing it , this null checks can help us from paged even check ( which is present in isPageRequestCreated() && isRequestChangedToPaged() method

Reply:
According to the [schema](https://github.com/folio-org/mod-remote-storage/blob/master/src/main/resources/swagger.api/schemas/pubSubEvent.json) every PubSubEvent should contain payload, not only PagedRequestCreated or RequestChangedToPaged. Are you sure that for PagedRequestCreated or RequestChangedToPaged events payload is always not null? What if for one of the mentioned requests payload is null? In this case isPagedRequestCreated or isRequestChangedToPaged methods will always return false and this is not correct.

Reply:
Yes, I see the log, but as per the information from RRT channel this issue occurred randomly and not only for checkout event as I remember. I mentioned checkout event in the Jira ticket just as example.

Reply:
However, if you are 100% sure that for request events payload is always not null, then OK. I just wanted to notice that if payload is null, then it does not mean that event is not request paged or request changed to paged. But this is your logic.